### ISSUE TITLE
Use roleTerm for labeling

### ISSUE THREAD

**Issue by drh-stanford** [sul-dlss/exhibits#936]
Broken out from #910.

Use <roleTerm> as labels in place of "Contributor" or "Author/Creator" labels.

Multiple people with the same role should be grouped under one label (ideally in a list!). The same person with multiple roles should be splatted out into multiple label/value pairs.

```
Contributor:
  John Doe (Lithographer)
```

would become:

```
Lithographer:
  John Doe
```

**Comment by drh-stanford** [comment]
@jvine @arcadiafalcone @blalbrit i have a concern about merging the contributor sections.

First, the current way (fs947tw3404) is displayed in the MODS modal for authors:

![screen shot 2017-11-22 at 12 28 31 pm](https://user-images.githubusercontent.com/1861171/33148601-4e507f70-cf81-11e7-819c-bbe27a9f6457.png)

Then, my changes to do the relabeling:

![screen shot 2017-11-22 at 12 30 27 pm](https://user-images.githubusercontent.com/1861171/33148620-5c85e256-cf81-11e7-807d-11e436debdcf.png)


Note that the contributors appear in the order that they appear in the MODS metadata. 

My concern is that we may want to preserve that ordering, or does the ordering not matter? There are 4 surveyor sections for example -- is preserving that sectioning important/meaningful?

Finally, the object show page has a very different display of the contributor metadata, for example:

![screen shot 2017-11-22 at 12 36 09 pm](https://user-images.githubusercontent.com/1861171/33148749-d94d3aaa-cf81-11e7-8787-e76800c3d1d0.png)

Those metadata come from our Solr index, not MODS. Do we need to do something else? Maybe that's an open question still?

**Comment by arcadiafalcone** [comment]
I would be comfortable with 1) lumping all the names with the same role under a single heading, 2) alphabetizing that group of names, and 3) dropping any exact duplicate names within the group (duplicates may be generated when converting from MARC).

The role groups should still appear in the order in which the first instance of each role appears in the MODS, as there is a general tendency to put the more-important-in-the-context-of-that-resource ones first.

That tendency of starting with the most important can apply to names as well, but I think in this case the usability gain of having the names grouped and ordered logically supersedes any informal expectation from metadata creators that the order of names will be retained in display.

**Comment by jvine** [comment]
I agree with grouping all the names with the same role under a single heading, and retaining the order of first appearance of that role. So in the sample above:
surveyor
cartographer
editor
electrotyper
lithographer

In the spirit of more-important-first, though, I'm not sure I agree about alphabetizing and de-duping those names. In this example, the 1st-listed surveyor would end up ~12 in the alpha list. There's a potentially usability gain in terms of being able to scan a long list for a specific name, and avoiding confusing duplicates (which seem like a data error we should handle somewhere else?) - but a loss in terms of understanding who was more responsible than the other guy. We get a lot of feedback around this in SearchWorks already.

As for the show page - I don't think that's in the scope of this issue at all, and we broadly accept that the show page metadata can be differently formatted and presented from the mods modal.

**Comment by arcadiafalcone** [comment]
My sense is that the most-important-first ordering happens just because human brains tend to think that way, and is not strictly applied. In the example record, the two last entries appear at the end because the MODS was converted from MARC and the 710 corporate name fields come after all the 700 personal name fields. I took a look at the resource itself, and the ordering is: all the names on the first chart in the order they appear, the names on the second chart that weren't on the first chart, and so on. Even with records not derived from MARC, sometimes it's just the order that made sense for data entry in the replayable spreadsheet rather than any judgment of relevant importance.

In short, I think there are enough other factors messing with the ordering that putting it in generally-comprehensible alphabetical order rather than a possibly-arbitrary order that is opaque to the user would be preferable. 

For the duplicates, yes, ideally that should be fixed in the MARC2MODS XSLT, but that is a whole 'nother kettle of worms. In any case, it's unlikely that we will do a mass reconversion with an updated XSLT, so if it's easy to catch it at display it would be a benefit to do so.

**Comment by jvine** [comment]
After offline discussion w @arcadiafalcone, here's the deal: 

- group all instances of a role together
- retain the order of first appearance of the roles
- retain order of appearance of names within each role

There will likely be further analysis/work on this down the road.

@drh-stanford are you enjoying this issue? 😄

**Comment by drh-stanford** [comment]
:) Ok, I think I have enough clarification to proceed, except one question. Which is what do I do when a person has multiple roles, like so?

```
      <mods>
        <name>
          <namePart>John Doe</namePart>
          <role>
            <roleTerm type='text'>Depicted</roleTerm>
          </role>
          <role>
            <roleTerm type='text'>Artist</roleTerm>
          </role>
        </name>
      </mods>
```

My understanding is that I would do this:

```
Depicted:
  John Doe

Artist:
  John Doe
```

and NOT

```
Depicted, Artist:
  John Doe
```

Am I understanding that correctly?

**Comment by jvine** [comment]
Yes, correct.

**Comment by drh-stanford** [comment]
@jvine @arcadiafalcone please review for accuracy, thanks!

Here's a screenshot of what I have thus far for the role consolidation. It's for the https://purl.stanford.edu/fs947tw3404 object.

![screen shot 2017-11-27 at 2 04 33 pm](https://user-images.githubusercontent.com/1861171/33291813-f3e89276-d37b-11e7-81c1-f40abe24a11c.png)

Note that Harrison appears twice in the MODS.

**Comment by jvine** [comment]
That looks great to me @drh-stanford

**Comment by caaster** [comment]
See also #2003
